tiledb/sm/cpp_api/query_experimental.h

```diff
@@ -236,6 +236,117 @@ class QueryExperimental {
     // Set the data buffer.
     return query.set_data_buffer(name, &data[0], data.size(), sizeof(char));
   }
+
+  /**
+   * Returns the number of elements in the result buffers from a read query.
+   * This is a map from the attribute name to a pair of values.
+   *
+   * The first is number of elements (offsets) for var size attributes, and the
+   * second is number of elements in the data buffer. For fixed sized attributes
+   * (and coordinates), the first is always 0.
+   *
+   * For variable sized attributes: the first value is the
+   * number of cells read, i.e. the number of offsets read for the attribute.
+   * The second value is the total number of elements in the data buffer. For
+   * example, a read query on a variable-length `float` attribute that reads
+   * three cells would return 3 for the first number in the pair. If the total
+   * amount of `floats` read across the three cells was 10, then the second
+   * number in the pair would be 10.
+   *
+   * For fixed-length attributes, the first value is always 0. The second value
+   * is the total number of elements in the data buffer. For example, a read
+   * query on a single `float` attribute that reads three cells would return 3
+   * for the second value. A read query on a `float` attribute with cell_val_num
+   * 2 that reads three cells would return 3 * 2 = 6 for the second value.
+   *
+   * If the query has not been submitted, an empty map is returned.
+   */
+  static std::unordered_map<std::string, std::pair<uint64_t, uint64_t>>
+  result_buffer_elements(const Query& query) {
```

Comment:
This seems like it's operating only on dimension labels? If so, the function name should reflect that. The comment should also be updated. Same below.